src/cls/rgw/cls_rgw_types.cc

```diff
@@ -577,6 +612,10 @@ void rgw_bucket_category_stats::dump(Formatter *f) const
   f->dump_unsigned("actual_size", actual_size);
 }
 
+void rgw_bucket_category_stats::decode_json(JSONObj *obj)
+{
```

Comment:
does this really have nothing to do?

Reply:
I am facing problem in the body of these functions, like how do these functions work. That's why I just added function definition and called it.

Reply:
Each of these structs has some data.  This data needs to be encoded, decoded, and dumped.    You can look at the struct's encode() and decode() functions to see what's important.  So in the case of rgw_bucket_category_stats, it has 4 fields, all uint64_t.  It would look something like this:

JSONDecoder::decode_json("total size", total_size, obj);
JSONDecode::decode_json("total size rounded", total_size_rounded, obj);
JSONDecode::decode_json("number entries", num_entries, obj);
JSONDecode::decode_json("actual size", actual_size, obj);

Note that each of those fields are encoded in rgw_bucket_category_stats::encode() (which is defined in the header).